### PR TITLE
ref(api): Move chunk data types to submodule

### PR DESCRIPTION
**Pull request by szokeasaurusrex**
- Create a new `data_types` submodule under the `api` module.
 - Crate a `chunking` submodule under the `data_types` module.
 - Move all data types related to chunking from `src/api/mod.rs` to the `chunking` submodule, splitting their definitions further into submodules under `chunking`.

This PR only moves existing code, adding `mod` and `use` statements and adjusting visibility as needed. No other modifications were made.

Ref #2077

### CODE REVIEW

File: src/api/data_types/chunking/compression.rs [hunk 1]
```diff
@@ -0,0 +1,48 @@
+use std::fmt;
+
+use serde::{Deserialize, Deserializer};
+
+#[derive(Debug, Clone, Copy, Ord, PartialOrd, Eq, PartialEq, Default)]
+pub enum ChunkCompression {
```
Comment (Swatinem) [inline src/api/data_types/chunking/compression.rs:6]: driveby question: how much effort would it be to add `zstd` compression to the whole workflow?

**Review by Swatinem — APPROVED** [review]

File: src/api/data_types/chunking/compression.rs (hunk 1 above)
Comment (szokeasaurusrex) [reply src/api/data_types/chunking/compression.rs:6]: I am not sure – have not worked too extensively with compression in the CLI yet.

What would be the benefit of adding `zstd`? Do we have any docs written about it?

Comment (Swatinem) [reply src/api/data_types/chunking/compression.rs:6]: no, we don’t have any docs on that.
and `zstd` is pretty much the best choice if you want to trade compression ratio for resource usage.

Comment (szokeasaurusrex) [reply src/api/data_types/chunking/compression.rs:6]: Ok, we should probably investigate how much effort this would be, will open an issue

**Comment by szokeasaurusrex** [comment]
@loewenheim Fair enough, I guess it comes down to personal preference a bit.

My reasoning is that each of these new modules has now a distinct responsibility; all code in a single module is closely related. When files get long, I have difficulty keeping track of everything in the file. It is much easier for me to work with many small files where I can clearly wrap my head around each one
